### PR TITLE
`uvx` warn when no executables are available

### LINKED ISSUE #5623: `uvx` should note if no executables are provided by a package
e.g.

```
❯ uvx requests
warning: `uvx` is experimental and may change without warning
Resolved 5 packages in 102ms
Installed 5 packages in 10ms
 + certifi==2024.7.4
 + charset-normalizer==3.3.2
 + idna==3.7
 + requests==2.32.3
 + urllib3==2.2.2
warning: An executable named `requests` is not provided by package `requests`.
The executable `requests` was not found.
```

I'm left guessing if `requests` happens to provide another executable.

### PR DESCRIPTION
**Pull request by blueraft**
## Summary

Resolves #5623 

## Test Plan

`cargo test`

### CODE REVIEW

File: crates/uv/src/commands/tool/run.rs [hunk 1]
```diff
@@ -153,12 +153,18 @@ pub(crate) async fn run(
...
-                        executable.to_string_lossy().cyan(),
-                    )?;
-                    if !entrypoints.is_empty() {
+                    if entrypoints.is_empty() {
+                        writeln!(
+                            printer.stdout(),
+                            "No executables, including `{}`, were found.",
+                            executable.to_string_lossy().cyan(),
+                        )?;
+                    } else {
```
Comment (zanieb) [inline crates/uv/src/commands/tool/run.rs:162]: I'd leave the above as-is and in the new message say something like "warning: `{from.name}` does not provide any executables". Wdyt?

Comment (blueraft) [reply crates/uv/src/commands/tool/run.rs:162]: Yeah makes sense, I will push the change in a sec.

Comment (blueraft) [reply crates/uv/src/commands/tool/run.rs]: Done

File: crates/uv/tests/tool_run.rs [hunk 2]
```diff
@@ -835,3 +836,33 @@ fn tool_run_without_output() {
...
+    Resolved 5 packages in [TIME]
+    Prepared 5 packages in [TIME]
+    Installed 5 packages in [TIME]
+     + certifi==2024.2.2
+     + charset-normalizer==3.3.2
+     + idna==3.6
+     + requests==2.31.0
+     + urllib3==2.2.1
+    warning: An executable named `requests` is not provided by package `requests`.
+    warning: `requests` does not provide any executables.
```
Comment (zanieb) [inline crates/uv/tests/tool_run.rs:866]: 🤔  We should probably display the second warning _instead_ of the first warning if we can.

Comment (blueraft) [reply crates/uv/tests/tool_run.rs]: Done

**Review by zanieb — APPROVED** [review]
Thank you!

**Pull request merged**